libnopegl/src/pgcraft.c

```diff
@@ -556,7 +556,7 @@ static int inject_block(struct pgcraft *s, struct bstr *b,
     for (int i = 0; i < ngli_darray_count(&block->fields); i++) {
         const struct block_field *fi = &field_info[i];
         const char *type = get_glsl_type(fi->type);
-        if (named_block->variadic && fi->count && i == ngli_darray_count(&block->fields))
+        if (fi->count == NGLI_BLOCK_VARIADIC_COUNT)
```

Comment:
I think parts of this change belongs in the previous commit (adding NGLI_BLOCK_VARIADIC_COUNT). In this commit, you should only drop `named_block->variadic && fi->count && i == ngli_darray_count(&block->fields)` condition.

Reply:
Mmh yeah sure makes sense, history adjusted.